reload/reload_test.go

```diff
@@ -0,0 +1,308 @@
+					Expect(require).To(Equal(true))
+				})
+			})
+
+			context("when BP_LIVE_RELOAD_ENABLED is '1'", func() {
+				it.Before(func() {
+					Expect(os.Setenv("BP_LIVE_RELOAD_ENABLED", "1")).To(Succeed())
+				})
+
+				it.After(func() {
+					Expect(os.Unsetenv("BP_LIVE_RELOAD_ENABLED")).To(Succeed())
+				})
+
+				it("should require watchexec at launch", func() {
+					require, err := reload.ShouldRequireWatchexec()
+					Expect(err).NotTo(HaveOccurred())
+					Expect(require).To(Equal(true))
+				})
+			})
+
+			context("when BP_LIVE_RELOAD_ENABLED is 'T'", func() {
+				it.Before(func() {
+					Expect(os.Setenv("BP_LIVE_RELOAD_ENABLED", "T")).To(Succeed())
+				})
+
+				it.After(func() {
+					Expect(os.Unsetenv("BP_LIVE_RELOAD_ENABLED")).To(Succeed())
+				})
+
+				it("should require watchexec at launch", func() {
+					require, err := reload.ShouldRequireWatchexec()
+					Expect(err).NotTo(HaveOccurred())
+					Expect(require).To(Equal(true))
+				})
+			})
+		})
+
+		context("when it should not enable live reload", func() {
+			context("when BP_LIVE_RELOAD_ENABLED is not set", func() {
+				it("should not require anything at launch", func() {
+					require, err := reload.ShouldRequireWatchexec()
+					Expect(require).To(Equal(false))
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+
+			context("when BP_LIVE_RELOAD_ENABLED is 'F'", func() {
+				it.Before(func() {
+					Expect(os.Setenv("BP_LIVE_RELOAD_ENABLED", "F")).To(Succeed())
+				})
+
+				it.After(func() {
+					Expect(os.Unsetenv("BP_LIVE_RELOAD_ENABLED")).To(Succeed())
+				})
+
+				it("should not require anything at launch", func() {
+					require, err := reload.ShouldRequireWatchexec()
+					Expect(require).To(Equal(false))
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+		})
+
+		context("failure cases", func() {
+			context("when BP_LIVE_RELOAD_ENABLED is not a valid boolean value", func() {
+				it.Before(func() {
+					Expect(os.Setenv("BP_LIVE_RELOAD_ENABLED", "hi")).To(Succeed())
+				})
+
+				it.After(func() {
+					Expect(os.Unsetenv("BP_LIVE_RELOAD_ENABLED")).To(Succeed())
+				})
+
+				it("should return an error", func() {
+					_, err := reload.ShouldRequireWatchexec()
+					Expect(err.Error()).To(Equal(`failed to parse BP_LIVE_RELOAD_ENABLED value hi: strconv.ParseBool: parsing "hi": invalid syntax`))
+				})
+			})
+		})
+	})
+
+	context("TransformReloadableProcesses", func() {
+		var (
+			originalProcess      packit.Process
+			watchExecProcessSpec reload.WatchExecProcessSpec
+
+			expectecNonReloadableProcess packit.Process
+			expectedReloadableProcess    packit.Process
+		)
+
+		it.Before(func() {
+			originalProcess = packit.Process{
+				Type:    "my-type",
+				Command: "my-command",
+				Args: []string{
+					"original-arg0",
+					"original-arg1",
+				},
+				Direct:           false,
+				Default:          false,
+				WorkingDirectory: "my-working-directory",
+			}
+			watchExecProcessSpec = reload.WatchExecProcessSpec{
```

Comment:
super small nit - but I would prefer to see the processes grouped together - and therefore i would move the watchexec spec to the top or bottom of the group of processes.

Reply:
I'm not sure what you mean - move the test `context`?

Reply:
no, sorry, just reorder the variable definitions so it looks like:

```
watchExecProcessSpec = ...
originalProcess = ...
expectedNonReloadableProcess = ...
expectedReloadableProcess = ...
```

Reply:
oh I see. Done